File: Web/Mvc/MvcRdlc/MvcRdlc/Pages/RdlcViewer.ascx.cs
namespace MvcRdlc.Pages
{
    using System;
    using System.Data;
    using System.Web.Mvc;

    using Microsoft.Reporting.WebForms;

    public partial class RdlcViewer
        : ViewUserControl
    {
        protected override void OnInit(
            EventArgs e)
        {
            base.OnInit(e);
            this.Context.Handler = this.Page;
        }

        protected override void OnLoad(
            EventArgs e)
        {
            base.OnLoad(e);

            if (this.IsPostBack)
            {
                return;
            }

            var xmlFile =
                this.MapPath(@"~\App_Data\Base.xml");

            var dataSet = new DataSet();
            dataSet.ReadXml(xmlFile);

            this.ReportViewer1.LocalReport.ReportPath =
                this.MapPath(@"~\App_Data\Base.rdlc");

            this.ReportViewer1.LocalReport.DataSources.Add(
                new ReportDataSource(
                    @"VSales",
                    dataSet.Tables[0]));
        }
    }
}

File: Web/Mvc/MvcRdlc/MvcRdlcVb/Pages/RdlcViewer.ascx.vb
Imports System.Web.Mvc
Imports Microsoft.Reporting.WebForms

Namespace Pages

    Public Class RdlcViewer
        Inherits ViewUserControl

        Protected Overrides Sub OnInit(e As EventArgs)
            MyBase.OnInit(e)
            Context.Handler = Page
        End Sub

        Protected Overrides Sub OnLoad(e As EventArgs)
            MyBase.OnLoad(e)

            If IsPostBack Then
                Return
            End If

            Dim xmlFile = MapPath("~\App_Data\Base.xml")
            Dim rptFile = MapPath("~\App_Data\Base.rdlc")

            Dim dataSet = New DataSet()
            dataSet.ReadXml(xmlFile)

            With ReportViewer1.LocalReport
                .ReportPath = rptFile
                .DataSources.Add( _
                    New ReportDataSource() With { _
                        .Name = "VSales", _
                        .Value = dataSet.Tables(0) _
                    })
            End With
        End Sub
    End Class
End Namespace